### PR TITLE
Add a shim for flexbox gap properties

### PR DESCRIPTION
**Pull request by devongovett**
This adds a shim for the flexbox gap properties so we can support them on the `Flex` component. They are currently implemented natively only in Firefox. We automatically detect this, and apply a shim if not. The shim is implemented by adding margins to each of the children, and counteracting this on a parent element.

### CODE REVIEW

File: packages/@react-spectrum/layout/src/flex.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+.flex-container {
+  /* this is necessary so that the inner margins don't affect anything outside */
+  overflow: hidden;
```
Comment (snowystinger) [inline packages/@react-spectrum/layout/src/flex.css:15]: This is going to show that problem with button focus rings pretty easily

File: packages/@react-spectrum/utils/src/styleProps.ts [hunk 2]
```diff
@@ -232,19 +232,26 @@ export function passthroughStyle(value) {
...
-  justifyItems: ['justifyItems', passthroughStyle],
-  justifyContent: ['justifyContent', passthroughStyle],
-  alignItems: ['alignItems', passthroughStyle],
-  alignContent: ['alignContent', passthroughStyle],
-  ...baseStyleProps
-};
+// Normalize 'start' and 'end' alignment values to 'flex-start' and 'flex-end'
+// in flex containers for browser compatibility.
+function flexAlignValue(value) {
+  if (value === 'start') {
```
Comment (snowystinger) [inline packages/@react-spectrum/utils/src/styleProps.ts:238]: any test we can do to see if 'start' is supported before changing it to 'flex-start'?

File: packages/@react-spectrum/layout/src/flex.css (hunk 1 above)
Comment (devongovett) [reply packages/@react-spectrum/layout/src/flex.css:15]: yeah, the problem is the inner div has a negative margin which affects the height of the outer div. not sure if there is another way to do this.

File: packages/@react-spectrum/utils/src/styleProps.ts (hunk 2 above)
Comment (devongovett) [reply packages/@react-spectrum/utils/src/styleProps.ts:238]: we could, but is it worth it?

**Pull request merged**